tests/test_salesforce_full_table_replication.py

```diff
@@ -0,0 +1,93 @@
+"""
+Test tap gets all records for streams with full replication
+"""
+import json
+
+from tap_tester import menagerie, runner, connections
+
+from base import SalesforceBaseTest
+
+class SalesforceFullReplicationTest(SalesforceBaseTest):
+    """Test tap gets all records for streams with full replication"""
+
+    @staticmethod
+    def name():
+        return "tt_salesforce_full_table"
+
+    def test_run(self):
+        """
+        Verify that the second sync includes the same number or more records than the first sync
+        Verify that all records in the first sync are included in the second sync
+        Verify that the sync only sent records to the target for selected streams (catalogs)
+
+        PREREQUISITE
+        For EACH stream that is fully replicated there are multiple rows of data with
+        """
+        
+        self.salesforce_api = 'BULK'
+        conn_id = connections.ensure_connection(self)
+        self.conn_id = conn_id
+
+        # Select all streams and no fields within streams
+        found_catalogs = self.run_and_verify_check_mode(conn_id)
+
+        # Test for streams that has the replication method as Full
+        full_streams = {'TabDefinition', 'FormulaFunctionAllowedType', 'FormulaFunction' }
+
+        our_catalogs = [catalog for catalog in found_catalogs if
+                        catalog.get('tap_stream_id') in full_streams]
+        self.select_all_streams_and_fields(conn_id, our_catalogs, select_all_fields=True)
+
+        # Run a sync job using orchestrator
+        first_sync_record_count = self.run_and_verify_sync(conn_id)
+
+        # verify that the sync only sent records to the target for selected streams (catalogs)
+        self.assertEqual(set(first_sync_record_count.keys()), full_streams)
+
+        first_sync_state = menagerie.get_state(conn_id)
+
+        # Get the set of records from a first sync
+        first_sync_records = runner.get_records_from_target_output()
+
+        # Run a second sync job using orchestrator
+        second_sync_record_count = self.run_and_verify_sync(conn_id)
+
+        # Get the set of records from a second sync
+        second_sync_records = runner.get_records_from_target_output()
+
+        for stream in full_streams.difference(self.child_streams()):
+            with self.subTest(stream=stream):
+
```

Comment:
I know it's also covered in the bookmarks test but since it is also specific to FULL_TABLE I'm in favor of adding these two assertions:
`                  # Verify full table streams do not save bookmarked values
                    # at the conclusion of a successful sync
                    self.assertIsNone(bookmark_value_1)
                    self.assertIsNone(bookmark_value_2)`

Reply:
bookmark_value_1 is obtained with the replication key. But for full streams, we don't have replication key. 
{'current_stream': None, 'bookmarks': {'FormulaFunctionAllowedType': {'version': None}, 'Formula
Function': {'version': None}, 'TabDefinition': {'version': None}}} 
This is how the state looks after the initial sync.

Reply:
Ok.. got it. Implemented this assertion as well.